### PR TITLE
Added new events to lock

### PR DESCRIPTION
**Pull request by tehsis**
This PR documents the new life cycle events in lock-web that were introduced by: https://github.com/auth0/lock/pull/196

### CODE REVIEW

File: articles/libraries/lock/events.md [hunk 1]
```diff
@@ -26,12 +26,24 @@ lock.removeListener('signin ready', signinHandlerFn);
...
+- `reset submit`: Triggers when reset panel is being submitted (for database and enterprise connections).
+- `reset success`: Triggers when reset has succeeded with no error.
+- `reset error`: Triggers when there's an error on the reset workflow.
 - `loggedin ready`: Triggers when loggedin mode is displayed.
+- `loggedin submit`: Triggers when loggedin panel is being submitted or the user clicks on a social connection.
 - `kerberos ready`: Triggers when integrated windows authentication mode is displayed.
+- `kerberos submit`: Triggers when integrated windows authentication mode is displayed.
 - `loading ready`: Triggers when loading mode is displayed.
+- `error shown`:  Triggers where an error was rendered into the widget.
+- `close`: Triggers when the user manually closes the widget.
```
Comment (cristiandouce) [inline articles/libraries/lock/events.md]: @tehsis please move this one to top... right before the `hidden` event please.

**Comment by cristiandouce** [comment]
@tehsis change `widget` to `Lock` instead? look all other references to the widget are as `Lock`.

With that we should be Ok for merge.

File: articles/libraries/lock/events.md [hunk 2]
```diff
@@ -24,14 +24,26 @@ lock.removeListener('signin ready', signinHandlerFn);
...
 - `signup ready`: Triggers when signup mode is displayed.
+- `signup submit`: Triggers when singup mode is being submitted or the user clicks on a social connection.
+- `signup success`: Triggers when signup was succeeded with no error.
+- `signup error`: Triggers when there's an error on the signup workflow..
 - `reset ready`: Triggers when reset mode is displayed.
+- `reset submit`: Triggers when reset mode is being submitted (for database and enterprise connections).
+- `reset success`: Triggers when reset has succeeded with no error.
+- `reset error`: Triggers when there's an error on the reset workflow.
 - `loggedin ready`: Triggers when loggedin mode is displayed.
+- `loggedin submit`: Triggers when loggedin panel is being submitted or the user clicks on a social connection.
```
Comment (cristiandouce) [inline articles/libraries/lock/events.md]: `s/is being/is/`

File: articles/libraries/lock/events.md [hunk 3]
```diff
@@ -24,14 +24,26 @@ lock.removeListener('signin ready', signinHandlerFn);
...
-- `signin submit`: Triggers when signin panel is being submitted (for database and enterprise connections).
+- `signin submit`: Triggers when signin mode is being submitted or the user clicks on a social connection.
+- `signin success`: Triggers when signin has succeeded with no error.
+- `signin error`: Triggers when there's an error on the signin workflow.
 - `signup ready`: Triggers when signup mode is displayed.
+- `signup submit`: Triggers when singup mode is being submitted or the user clicks on a social connection.
+- `signup success`: Triggers when signup was succeeded with no error.
+- `signup error`: Triggers when there's an error on the signup workflow..
 - `reset ready`: Triggers when reset mode is displayed.
+- `reset submit`: Triggers when reset mode is being submitted (for database and enterprise connections).
```
Comment (cristiandouce) [inline articles/libraries/lock/events.md]: `s/is being/is/`

File: articles/libraries/lock/events.md [hunk 4]
```diff
@@ -24,14 +24,26 @@ lock.removeListener('signin ready', signinHandlerFn);
...
-- `hidden`: Triggers when Lock has hidden.
+- `close`: Triggers when the user manually closes the Lock.
+- `hidden`: Triggers when the Lock has hidden.
 - `signin ready`: Triggers when signin mode view is displayed.
-- `signin submit`: Triggers when signin panel is being submitted (for database and enterprise connections).
+- `signin submit`: Triggers when signin mode is being submitted or the user clicks on a social connection.
+- `signin success`: Triggers when signin has succeeded with no error.
+- `signin error`: Triggers when there's an error on the signin workflow.
 - `signup ready`: Triggers when signup mode is displayed.
+- `signup submit`: Triggers when singup mode is being submitted or the user clicks on a social connection.
```
Comment (cristiandouce) [inline articles/libraries/lock/events.md]: `s/is being/is/`

File: articles/libraries/lock/events.md [hunk 5]
```diff
@@ -24,14 +24,26 @@ lock.removeListener('signin ready', signinHandlerFn);
...
+- `reset submit`: Triggers when reset mode is being submitted (for database and enterprise connections).
+- `reset success`: Triggers when reset has succeeded with no error.
+- `reset error`: Triggers when there's an error on the reset workflow.
 - `loggedin ready`: Triggers when loggedin mode is displayed.
+- `loggedin submit`: Triggers when loggedin panel is being submitted or the user clicks on a social connection.
 - `kerberos ready`: Triggers when integrated windows authentication mode is displayed.
+- `kerberos submit`: Triggers when integrated windows authentication mode is submitted.
 - `loading ready`: Triggers when loading mode is displayed.
+- `error shown`:  Triggers where an error was rendered into the Lock.
 - `client initialized`: (**internal only**)Triggers when `clientID`'s config data is fetched and loaded.
```
Comment (cristiandouce) [inline articles/libraries/lock/events.md]: will you also move this ti the `Internals` section and also add the ones missing?
I think those are `client load error` and `client load success`...

File: articles/libraries/lock/events.md [hunk 6]
```diff
@@ -24,15 +24,26 @@ lock.removeListener('signin ready', signinHandlerFn);
...
+- `hidden`: Triggers when the Lock has hidden.
 - `signin ready`: Triggers when signin mode view is displayed.
-- `signin submit`: Triggers when signin panel is being submitted (for database and enterprise connections).
+- `signin submit`: Triggers when signin mode is submitted.
+- `signin success`: Triggers when signin has succeeded with no error.
+- `signin error`: Triggers when there's an error on the signin workflow.
 - `signup ready`: Triggers when signup mode is displayed.
+- `signup submit`: Triggers when singup mode is submitted.
+- `signup success`: Triggers when signup was succeeded with no error.
+- `signup error`: Triggers when there's an error on the signup workflow..
```
Comment (cristiandouce) [inline articles/libraries/lock/events.md]: double `..` on the end

File: articles/libraries/lock/events.md [hunk 7]
```diff
@@ -24,15 +24,26 @@ lock.removeListener('signin ready', signinHandlerFn);
...
+- `reset submit`: Triggers when reset mode is submitted (for database and enterprise connections).
+- `reset success`: Triggers when reset has succeeded with no error.
+- `reset error`: Triggers when there's an error on the reset workflow.
 - `loggedin ready`: Triggers when loggedin mode is displayed.
+- `loggedin submit`: Triggers when loggedin panel is submitted.
 - `kerberos ready`: Triggers when integrated windows authentication mode is displayed.
+- `kerberos submit`: Triggers when integrated windows authentication mode is submitted.
 - `loading ready`: Triggers when loading mode is displayed.
-- `client initialized`: (**internal only**)Triggers when `clientID`'s config data is fetched and loaded.
+- `error shown`:  Triggers when an error was displayed into the Lock.
```
Comment (cristiandouce) [inline articles/libraries/lock/events.md]: extra space after `:` ?
remove the `into the Lock`... end it with `displayed.`

File: articles/libraries/lock/events.md [hunk 8]
```diff
@@ -24,15 +24,26 @@ lock.removeListener('signin ready', signinHandlerFn);
...
-- `signin submit`: Triggers when signin panel is being submitted (for database and enterprise connections).
+- `signin submit`: Triggers when signin mode is submitted.
+- `signin success`: Triggers when signin has succeeded with no error.
+- `signin error`: Triggers when there's an error on the signin workflow.
 - `signup ready`: Triggers when signup mode is displayed.
+- `signup submit`: Triggers when singup mode is submitted.
+- `signup success`: Triggers when signup was succeeded with no error.
+- `signup error`: Triggers when there's an error on the signup workflow.
 - `reset ready`: Triggers when reset mode is displayed.
+- `reset submit`: Triggers when reset mode is submitted (for database and enterprise connections).
```
Comment (cristiandouce) [inline articles/libraries/lock/events.md]: the `(for database and enterprise connections)` was from signin... why is it in `reset`? I think this is missplaced.

Comment (tehsis) [inline articles/libraries/lock/events.md]: We don't have reset on database and enterprise connection... probably is redundant.

Comment (cristiandouce) [inline articles/libraries/lock/events.md]: this is a copy/paste wrongly editted, it is actually redundant. If you check the diff you removed this comment from the `signin submit` event :dancers:

**Pull request merged**